tensorflow_io/bigquery/README.md

```diff
@@ -85,6 +85,81 @@ if __name__ == '__main__':
 
 ```
 
+It also supports reading repeated mode BigQuery column (each field contains array of values). In this case, selected_fields needs be a dictionary in a
+form like
+
+```
+        { "field_a_name": {"mode": "repeated", output_type: dtypes.int64},
+          "field_b_name": {"mode": "nullable", output_type: dtypes.string},
+          ...
+          "field_x_name": {"mode": "repeated", output_type: dtypes.string}
+        }
+
+```
+"mode" is BigQuery column attribute concept, it can be 'repeated', 'nullable' or 'required'.The output field order is unrelated to the order of fields in
```

Comment:
could you test that 'nullable' and 'required' work for DataFormat.ARROW and mention that 'repeated' only works for primitive types on leaf level fields for  DataFormat.AVRO

Reply:
> could you test that 'nullable' and 'required' work for DataFormat.ARROW

it looks like there is no existing unit test in tests/test_bigquery_eager.py for DataFormat.ARROW (https://github.com/tensorflow/io/pull/939/files and I also searched the repo).  It might not ideal to coupling adding a unit test for DataFormat.ARROW with this change, maybe better do it in a separate pull request? wdyt? 

I tried something like 

```
    def test_read_rows_nonrepeated_only_with_arrow(self):
        """Test for reading rows with non-repeated fields only, with DataFormat.ARROW, then selected_fields and output_types are list (backward compatible)."""
        client = BigQueryTestClient(BigqueryOpsTest.server.endpoint())
        read_session = self._get_read_session(client, nonrepeated_only=True, data_format=BigQueryClient.DataFormat.ARROW)

        streams_list = read_session.get_streams()
        self.assertEqual(len(streams_list), 2)
        dataset1 = read_session.read_rows(streams_list[0])
        itr1 = iter(dataset1)
        self.assertEqual(
            self._get_nonrepeated_only_fields(self.STREAM_1_ROWS[0]),
            self._normalize_dictionary(itr1.get_next()),
        )
        self.assertEqual(
            self._get_nonrepeated_only_fields(self.STREAM_1_ROWS[1]),
            self._normalize_dictionary(itr1.get_next()),
        )
        with self.assertRaises(errors.OutOfRangeError):
            itr1.get_next()

        dataset2 = read_session.read_rows(streams_list[1])
        itr2 = iter(dataset2)
        self.assertEqual(
            self._get_nonrepeated_only_fields(self.STREAM_2_ROWS[0]),
            self._normalize_dictionary(itr2.get_next()),
        )
        self.assertEqual(
            self._get_nonrepeated_only_fields(self.DEFAULT_VALUES),
            self._normalize_dictionary(itr2.get_next()),
        )
        with self.assertRaises(errors.OutOfRangeError):
            itr2.get_next()
```
and I am getting some issues

```
__________________________________________________________________________________________________________ BigqueryOpsTest.test_read_rows_nonrepeated_only_with_arrow __________________________________________________________________________________________________________

self = <tests.test_bigquery_eager.BigqueryOpsTest testMethod=test_read_rows_nonrepeated_only_with_arrow>

    def test_read_rows_nonrepeated_only_with_arrow(self):
        """Test for reading rows with non-repeated fields only, with DataFormat.ARROW, then selected_fields and output_types are list (backward compatible)."""
        client = BigQueryTestClient(BigqueryOpsTest.server.endpoint())
>       read_session = self._get_read_session(client, nonrepeated_only=True, data_format=BigQueryClient.DataFormat.ARROW)

tests/test_bigquery_eager.py:495:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/test_bigquery_eager.py:356: in _get_read_session
    data_format=data_format
tensorflow_io/bigquery/python/ops/bigquery_api.py:183: in read_session
    row_restriction=row_restriction,
<string>:177: in io_big_query_read_session
    ???
/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/ops.py:6653: in raise_from_not_ok_status
    six.raise_from(core._status_to_exception(e.code, message), None)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

value = None, from_value = None

>   ???
E   tensorflow.python.framework.errors_impl.InvalidArgumentError: AVRO schema is missing [Op:IO>BigQueryReadSession]

<string>:3: InvalidArgumentError
```
i think we need an ARROW schema in our test

> mention that 'repeated' only works for primitive types on leaf level fields for DataFormat.AVRO

Done

Reply:
You are right, I didn't add ARROW test yet, need to setup test infrastructure to support ARROW, so test is not expected to work. Could you just try it on any existing BQ dataset.

Reply:
Tried 

```
from tensorflow.python.framework import ops
from tensorflow.python.framework import dtypes
from tensorflow_io.bigquery import BigQueryClient
from tensorflow_io.bigquery import BigQueryReadSession

GCP_PROJECT_ID = '<Ruhua's_project_id>'
DATASET_GCP_PROJECT_ID = "bigquery-public-data"
DATASET_ID = "samples"
TABLE_ID = "wikipedia"

def main():
  ops.enable_eager_execution()
  client = BigQueryClient()
  read_session = client.read_session(
      "projects/" + GCP_PROJECT_ID,
      DATASET_GCP_PROJECT_ID, TABLE_ID, DATASET_ID,
      selected_fields = ["title"],
      output_types = [dtypes.string],
      requested_streams=2,
      row_restriction="num_characters > 1000",
      data_format=BigQueryClient.DataFormat.ARROW)
  dataset = read_session.parallel_read_rows()

  row_index = 0
  for row in dataset.prefetch(10):
    print("row %d: %s" % (row_index, row))
    row_index += 1

if __name__ == '__main__':
  main()
```
with master branch by following https://github.com/tensorflow/io

```
bazel build -s --verbose_failures //tensorflow_io/...
TFIO_DATAPATH=bazel-bin python3 -m pip install .
TFIO_DATAPATH=bazel-bin python3  <file contains above_code>.py
```
I was getting 

```
2020-07-07 16:26:28.394491: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/eager/context.py", line 1986, in execution_mode
    yield
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/data/ops/iterator_ops.py", line 655, in _next_internal
    output_shapes=self._flat_output_shapes)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/ops/gen_dataset_ops.py", line 2363, in iterator_get_next
    _ops.raise_from_not_ok_status(e, name)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/ops.py", line 6653, in raise_from_not_ok_status
    six.raise_from(core._status_to_exception(e.code, message), None)
  File "<string>", line 3, in raise_from
tensorflow.python.framework.errors_impl.InternalError: Invalid: Invalid argument: arrow data type 0x7f1a48016fa0 is not supported: Type error: Arrow data type is not supported [Op:IteratorGetNext]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "test_bigquery_eager.py", line 66, in <module>
    main()
  File "test_bigquery_eager.py", line 61, in main
    for row in dataset.prefetch(10):
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/data/ops/iterator_ops.py", line 631, in __next__
    return self.next()
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/data/ops/iterator_ops.py", line 670, in next
    return self._next_internal()
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/data/ops/iterator_ops.py", line 661, in _next_internal
    return structure.from_compatible_tensor_list(self._element_spec, ret)
  File "/usr/lib/python3.6/contextlib.py", line 99, in __exit__
    self.gen.throw(type, value, traceback)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/eager/context.py", line 1989, in execution_mode
    executor_new.wait()
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/eager/executor.py", line 67, in wait
    pywrap_tfe.TFE_ExecutorWaitForAllPendingNodes(self._handle)
tensorflow.python.framework.errors_impl.InternalError: Invalid: Invalid argument: arrow data type 0x7f1a48016fa0 is not supported: Type error: Arrow data type is not supported
```

Reply:
ah, nvm, i think this was due to arrow can't work with dtypes.string

replace 
```
      selected_fields = ["title"],
      output_types = [dtypes.string],
```
with 
```
      selected_fields = ["id"],
      output_types = [dtypes.int64],
```
things works perfectly fine with both master branch and my branch.

Also, as you suggested, test nullable and required with arrow in my branch

```
from tensorflow.python.framework import ops
from tensorflow.python.framework import dtypes
from tensorflow_io.bigquery import BigQueryClient
from tensorflow_io.bigquery import BigQueryReadSession

GCP_PROJECT_ID = 'twttr-cortex-mlengine-evl'
DATASET_GCP_PROJECT_ID = "bigquery-public-data"
DATASET_ID = "samples"
TABLE_ID = "wikipedia"

def main():
  ops.enable_eager_execution()
  client = BigQueryClient()
  read_session = client.read_session(
      "projects/" + GCP_PROJECT_ID,
      DATASET_GCP_PROJECT_ID, TABLE_ID, DATASET_ID,
      selected_fields = {
          "id": {"mode": BigQueryClient.FieldMode.NULLABLE, "output_type": dtypes.int64},
          "wp_namespace": {"mode": BigQueryClient.FieldMode.REQUIRED, "output_type": dtypes.int64},
          },
      requested_streams=2,
      row_restriction="num_characters > 1000",
      data_format=BigQueryClient.DataFormat.ARROW)
  dataset = read_session.parallel_read_rows()

  row_index = 0
  for row in dataset.prefetch(10):
    print("row %d: %s" % (row_index, row))
    row_index += 1

if __name__ == '__main__':
  main()
```
works great!

```
row 3829: OrderedDict([('id', <tf.Tensor: shape=(), dtype=int64, numpy=7863732>), ('wp_namespace', <tf.Tensor: shape=(), dtype=int64, numpy=0>)])
row 3830: OrderedDict([('id', <tf.Tensor: shape=(), dtype=int64, numpy=9317673>), ('wp_namespace', <tf.Tensor: shape=(), dtype=int64, numpy=0>)])
row 3831: OrderedDict([('id', <tf.Tensor: shape=(), dtype=int64, numpy=5986>), ('wp_namespace', <tf.Tensor: shape=(), dtype=int64, numpy=0>)])
row 3832: OrderedDict([('id', <tf.Tensor: shape=(), dtype=int64, numpy=15843197>), ('wp_namespace', <tf.Tensor: shape=(), dtype=int64, numpy=0>)])
```

Reply:
Perfect, thanks for your contribution!!!